cli/flox-rust-sdk/src/models/environment/path_environment.rs

```diff
@@ -479,57 +482,36 @@ impl PathEnvironment {
         Self::open(flox, pointer, dot_flox_path, temp_dir)
     }
 
-    /// Replace all placeholders in the manifest file contents
-    ///
-    /// TODO: we should probably be using toml_edit
-    fn replace_placeholders(
-        contents: &String,
+    /// Replace all placeholders in the `manifest.toml` file
+    fn replace_manifest_template_placeholders<'a>(
```

Comment:
This could be ideal, a reason we could argue against this is that `toml_edit` does not seem to support comment manipulation, it just preserves them.

Having a template string could allow us to define comments there, which could be helpful to end users.

Reply:
`toml_edit` does provide comment manipulation, they call it `Decor`:

```rust
let mut doc: DocumentMut = _;
let doc_decor = doc.decor_mut();
doc_decor.set_prefix("...");   /// should set the comments at the head of the document


let mut install_table = Table::new();
let table_decor = install_table.decor_mut();
install_key_decor.set_prefix("...");   /// a) decor above the `install` table 

doc.insert_formatted("install", install_table);


// b) maybe this is required to get the comment in the right place alternative to a)
doc.key_mut("install").unwrap().leaf_decor_mut().set_prefix("...")
```

Check the docs on Table there's a couple deprecated methods around decor usages

Reply:
Ah, nice, didn't know about that!

I searched for "comment" and nothing showed up, then assumed comments weren't supported. Thanks!